rust/private/rust.bzl

```diff
@@ -938,3 +938,15 @@ rust_benchmark(
 Run the benchmark test using: `bazel run //fibonacci:fibonacci_bench`.
 """,
 )
+
+def name_to_crate_name(name):
+    """Converts a build target's name into the name of its associated crate.
+
+    Crate names cannot contain certain characters, such as -, which are allowed
+    in build target names. All illegal characters will be converted to
+    underscores.
+
+    This is a similar conversion as that which cargo does, taking a
+    `Cargo.toml`'s `package.name` and canonicalizing it
+    """
```

Comment:
nit: Can you add `Arg` and `Return` sections with type hints 😅 (this one I won't fight anyone over but would still like to see it)

Reply:
Done :)